### PR TITLE
Fix varz_jetstream_meta_leader reports more than one leader

### LINKED ISSUE #179: varz_jetstream_meta_leader reports more than one leader
Hello :wave: 

We noticed the metrics `nats_varz_jetstream_meta_leader` reports multiple leaders at the same time. 

```
# HELP nats_varz_jetstream_meta_leader jetstream_meta_leader
# TYPE nats_varz_jetstream_meta_leader gauge
nats_varz_jetstream_meta_leader{server_id="NALM6ASRQ6HD645QEJRTVPRMIR2ENKXLYFFZFWP2YJIHIIHLMN235SNY",value="dev-aws-euw2-int-jetstream-0"} 1
nats_varz_jetstream_meta_leader{server_id="NALM6ASRQ6HD645QEJRTVPRMIR2ENKXLYFFZFWP2YJIHIIHLMN235SNY",value="dev-aws-euw2-int-jetstream-2"} 1
nats_varz_jetstream_meta_leader{server_id="NALM6ASRQ6HD645QEJRTVPRMIR2ENKXLYFFZFWP2YJIHIIHLMN235SNY",value="dev-eqix-ld4-int-jetstream-0"} 1
```

After restarting the pod, the metric was fixed and only one leader appeared in the query. I believe the old leader is not removed from the metrics when a new one is elected. 

Thanks :bowing_man:

### PR DESCRIPTION
**Pull request by VictorPerezForm3**
Fix #179

### CODE REVIEW

File: collector/collector.go [hunk 1]
```diff
@@ -98,8 +98,8 @@ func newLabelGauge(system, subsystem, name, help, prefix, label string) *prometh
 		Help:      help,
 	}
 	metric := prometheus.NewGaugeVec(opts, []string{"server_id", label})
-
-	Tracef("Created metric: %s, %s, %s, %s", namespace, subsystem, name, help)
+	metric.
```
Comment (derekcollison) [inline collector/collector.go:101]: This intentional?

File: collector/collector.go [hunk 2]
```diff
@@ -231,6 +231,7 @@ func (nc *NATSCollector) collectStatsFromRequests(
 			case float64: // json only has floats
 				m.WithLabelValues(id).Set(v)
 			case string:
+				m.Reset()
```
Comment (derekcollison) [inline collector/collector.go:234]: Nice catch!

File: collector/collector.go (hunk 1 above)
Comment (VictorPerezForm3) [reply collector/collector.go:101]: Nope! sorry, I think I just hit enter in the wrong place :sweat_smile:

**Comment by VictorPerezForm3** [comment]
After thinking about this issue, I believe the same thing will happen with float values because they contain dynamic labels such as "meta_leader". For example, for the metric `nats_stream_last_seq` In case of a meta leader relection, the value will freeze and a new metric will be created. 

I didn't attempt to reproduce it, but if you can reproduce, maybe it should just reset all gauges before setting the new values. This way old labels will be clean up. 

The problem is that this approach won't work for counters. Maybe you should think about removing dynamic labels from metrics. You can create a single gauge to hold all the values and then merge with other metrics if you need to extend the information.

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**